test/runtests.jl

```diff
@@ -82,3 +82,30 @@ for prob in names(ADNLPProblems)
     end
   end
 end
+
+function is_feasible(nlp::AbstractNLPModel; x = nlp.meta.x0)
+  feasible = missing
+  if get_ncon(nlp) == 0
+    feasible = true
+  else
+    c = cons(nlp, x)
+    if all(get_lcon(nlp) .<= c .<= get_ucon(nlp)) && all(get_lvar(nlp) .<= x .<= get_uvar(nlp))
+      feasible = true
+    end
+    if any(get_lvar(nlp) .> get_uvar(nlp)) || any(get_lcon(nlp) .> get_ucon(nlp))
+      feasible = false
+    end
+  end
+  return feasible
+end
+
+names_pb_vars = meta[meta.variable_nvar .== true, [:nvar, :name, :best_known_upper_bound, :best_known_lower_bound]]
+adproblems = (eval(Meta.parse("ADNLPProblems.$(pb[:name])()")) for pb in eachrow(names_pb_vars))
+for (pb, nlp) in zip(eachrow(names_pb_vars), adproblems)
+  @test pb[:nvar] == nlp.meta.nvar
+  if is_feasible(nlp) && get_minimize(nlp)
+    @test pb[:best_known_upper_bound] ≈ obj(nlp, nlp.meta.x0)
+  elseif is_feasible(nlp) && !get_minimize(nlp)
+    @test pb[:best_known_lower_bound] ≈ obj(nlp, nlp.meta.x0)
+  end
```

Comment:
These will fail for manually updated problems, right?

Reply:
Ouch, thanks for catching that. That was for the local test only.